src/tests/functional/frame_tests.ts

```diff
@@ -75,6 +75,17 @@ export class FrameTests extends TurboDriveTestCase {
     this.assert.notOk(await this.hasSelector("#nested-child"))
   }
 
+  async "test following a form within a nested frame with target top"() {
```

Comment:
A test for a `form` with `data-turbo-frame="_top"` might also be helpful here: https://turbo.hotwired.dev/reference/frames#frame-with-overwritten-navigation-targets